### PR TITLE
Emulator changes

### PR DESCRIPTION
**Pull request by lauracabayol**
Some new changes to the previous version in the input_emulator branch:
- Add the option of drop_z to the archive. Please @jchavesmontero check the base_archive changes I introduced. 
- Add option drop_sim and drop_z to the emulators
- Clean the obtain_params emulator function. Now it does not return Nz, but it was not used and as pointed out by Andreu, it was dangerous for simulations with missing snapshots
- In the same function, create on mask in k per snapshot since there are simulations with different number of k's. Use this mask instead of a hardcoded Nk option that was causing problems when running with Nyx simulations

### CODE REVIEW

File: lace/archive/nyx_archive.py [hunk 1]
```diff
@@ -41,7 +41,9 @@ def __init__(self, file_name=None, kp_Mpc=0.7):
         # list all simulations
         self.list_sim = self.list_sim_cube + self.list_sim_test
         ## done set simulation list
-
+        
+        # list of available redshifts, temporary hack, chech higher redshfits
+        self.list_sim_redshifts = [str(d) for d in np.round(np.arange(2.2,4.7,0.2),2) ]
```
Comment (andreufont) [inline lace/archive/nyx_archive.py:46]: @lauracabayol - this hard-coded list of redshifts is not great, since some sims have z=2.0 and most of them (all?) also go beyond z=4.6... do you need it?

Comment (andreufont) [reply lace/archive/nyx_archive.py:46]: I see that you are aware that this is a hack, but do you want to put it to the main branch?

File: lace/emulator/nn_emulator.py [hunk 2]
```diff
@@ -262,26 +268,20 @@ def _obtain_sim_params(self):
...
-        k_Mpc_train = k_Mpc[k_mask]
+        
+        k_mask = [
+            (self.training_data[i]["k_Mpc"] < self.kmax_Mpc) & (self.training_data[i]["k_Mpc"] > 0)
+            for i in range(len(self.training_data))
+        ]
+        
+        self.k_mask= k_mask
+        
+        k_Mpc_train = self.training_data[0]["k_Mpc"][k_mask[0]]
```
Comment (andreufont) [inline lace/emulator/nn_emulator.py:279]: I see that you are using the first sim in the training data. Are we sure that the k-range is the same for all the sims?

**Review by andreufont — COMMENTED** [review]
I added a couple of minor comments for @lauracabayol , but I don't see any major show stopper

File: lace/archive/nyx_archive.py (hunk 1 above)
Comment (andreufont) [reply lace/archive/nyx_archive.py:46]: For now, we can limit the Nyx archive to this redshift range, and only go outside this range after Durham.

File: lace/emulator/nn_emulator.py (hunk 2 above)
Comment (lauracabayol) [reply lace/emulator/nn_emulator.py:279]: Some sims have an additional k bin, but to k <4 all k are the same

File: lace/archive/nyx_archive.py (hunk 1 above)
Comment (andreufont) [reply lace/archive/nyx_archive.py:47]: I made some changes to cover all possible Nyx redshifts, from z=2 to 5.4, skipping 4.8 and 5.2

**Review by andreufont — APPROVED** [review]
I've tested LaCE and cup1d notebooks, I think we are ready to go. Waiting for Laura to test the speed of Nyx reading before merging
